### PR TITLE
WIP: Just to get eerhardt's thoughts

### PR DESCRIPTION
**Pull request by pgovind**
Do not merge. This is just to get some eyes on this idea.

### CODE REVIEW

File: src/Microsoft.Data/PrimitiveColumnContainer.cs [hunk 1]
```diff
@@ -14,7 +16,7 @@ namespace Microsoft.Data
     /// PrimitiveDataFrameColumnContainer is just a store for the column data. APIs that want to change the data must be defined in PrimitiveDataFrameColumn
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    internal partial class PrimitiveColumnContainer<T> : IEnumerable<T?>
+    internal partial class PrimitiveColumnContainer<T> : IEnumerable<ReadOnlyMemory<T>>
```
Comment (pgovind) [inline src/Microsoft.Data/PrimitiveColumnContainer.cs]: I'm not sure if this should be a `IEnumerable<ReadOnlyMemory<T>>` or a `IEnumerable<Memory<T>>`. A `PrimitiveColumnContainer` could hold either `ReadOnlyMemory<T>` or `Memory<T>` in its `Buffers`. `IEnumerable<ReadOnlyMemory<T>>` would work for either, but it'd be immutable. If we instead always expose an `IEnumerable<Memory<T>>`, it'd involve an unnecessary copy for simple enumeration use cases. 

Another option is to not touch the interface but add 2 new APIs that return a `IEnumerable<ReadOnlyMemory<T>>` and an `IEnumerable<Memory<T>>`. That way, the users of the API have options.

File: src/Microsoft.Data/StringColumn.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Data
     /// A mutable column to hold strings
     /// </summary>
     /// <remarks> Is NOT Arrow compatible </remarks>
-    public partial class StringColumn : BaseColumn, IEnumerable<string>
+    public partial class StringColumn : BaseColumn, IEnumerable<List<string>>
```
Comment (pgovind) [inline src/Microsoft.Data/StringColumn.cs]: Same alternative here. We could leave the interface as it is and make a new API that returned a `List<string>`. We'd have to deal with keeping the `NullCount` in sync, but that is easily solved

File: src/Microsoft.Data/PrimitiveColumn.cs [hunk 3]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.Data
     /// A column to hold primitive types such as int, float etc.
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public partial class PrimitiveColumn<T> : BaseColumn, IEnumerable<T?>
```
Comment (eerhardt) [inline src/Microsoft.Data/PrimitiveColumn.cs]: You need to still keep `IEnumerable<T?>`. This way things like XPlot charting still work by just passing it a Column, and it enumerates over the data in the column.

File: src/Microsoft.Data/PrimitiveColumnContainer.cs (hunk 1 above)
Comment (eerhardt) [reply src/Microsoft.Data/PrimitiveColumnContainer.cs]: Honestly, I think a new API (or two) is called for here. Something like `GetBuffers()` and `GetReadOnlyBuffers()`?

---
In reply to: [321517883](https://github.com/dotnet/corefxlab/pull/2718#discussion_r321517883) [](ancestors = 321517883)

**Pull request closed without merging**